agent/lib/kontena/websocket_client.rb

```diff
@@ -177,5 +182,28 @@ def verify_connection
     rescue => exc
       error exc.message
     end
+
+    # Abort the connection, closing the websocket, with a timeout
+    def close
+      return if @close_timer
+
+      # stop sending messages, queue them up until reconnected
+      Celluloid::Notifications.publish('websocket:disconnect', nil)
+
+      # send close frame; this will get stuck if the server is not replying
+      ws.close(1000)
+
+      @close_timer = EM::Timer.new(CLOSE_TIMEOUT) do
```

Comment:
If we upgrade to a faye-websocket release that includes https://github.com/faye/faye-websocket-ruby/commit/07be5a24ef446db5d7e461ca1eaa9fa6aa7b7f21, then there will be two concurrent timeouts...

Perhaps we should detect the presence of a `Faye::Websocket::Client.CLOSE_TIMEOUT`? We could still keep this timeout, but just make sure that it's longer than the faye close timeout?